### PR TITLE
Adds add_comment method to add text comments to existing Issues

### PR DESCRIPTION
**Pull request by eggmaster**
Adds very basic comment functionality. Comments can only be added to an existing Issue. Wraps any backend exceptions in a new exception. Moves exceptions to a separate file following the pattern in other projects.

### CODE REVIEW

File: tests/unit/test_bugjira.py [hunk 1]
```diff
@@ -180,6 +182,94 @@ def test_get_issue_with_non_string_key(sandboxed_bugjira):
...
+
+def test_add_comment_bugzilla_exception(sandboxed_bugjira):
+    """
+    GIVEN a Bugjira instance whose bugzilla backend's update_bugs method
+        raises an Exception
+    WHEN we call add_comment
+    THEN a BrokerAddCommentException should be raised
+    """
+    def raise_exception(*args):
+        raise Exception("Exception raised")
```
Comment (jpichon) [inline tests/unit/test_bugjira.py]: You don't need to do this here, you should be able to leverage '[side_effect](https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock.side_effect)' on the next line:

    sandboxed_bugjira.bugzilla.update_bugs.side_effect = Exception('Exception raised')

File: tests/unit/test_bugjira.py [hunk 2]
```diff
@@ -180,6 +182,94 @@ def test_get_issue_with_non_string_key(sandboxed_bugjira):
...
+    with pytest.raises(BrokerAddCommentException,
+                       match="Exception raised"):
+        sandboxed_bugjira.add_comment(issue, comment_text)
+
+
+def test_add_comment_good_jira(sandboxed_bugjira):
+    """
+    GIVEN a Bugjira instance
+    WHEN we call add_comment with a JiraIssue and comment string
+    THEN the bugzilla backend's build_update and update_bugs methods should
```
Comment (jpichon) [inline tests/unit/test_bugjira.py]: build_update doesn't seem to be checked here, either the docstring or the test should be tweaked.

File: tests/unit/test_bugjira.py [hunk 3]
```diff
@@ -180,6 +182,94 @@ def test_get_issue_with_non_string_key(sandboxed_bugjira):
...
+def test_add_comment_jira_exception(sandboxed_bugjira):
+    """
+    GIVEN a Bugjira instance whose jira backend's add_comment method
+        raises an Exception
+    WHEN we call add_comment
+    THEN a BrokerAddCommentException should be raised
+    """
+    def raise_exception(*args):
+        raise Exception("Exception raised")
+    sandboxed_bugjira.jira.add_comment = raise_exception
```
Comment (jpichon) [inline tests/unit/test_bugjira.py]: Same comment as above on side-effects.

**Review by jpichon — COMMENTED** [review]
A couple of very minor comments inline, LGTM otherwise!

**Review by jpichon — DISMISSED** [review]
LGTM!

File: tests/unit/test_bugjira.py (hunk 2 above)
Comment (eggmaster) [reply tests/unit/test_bugjira.py]: thank you for noticing my bad copy/paste!

Comment (jpichon) [reply tests/unit/test_bugjira.py]: Always happy to be of service ;D

File: src/bugjira/broker.py [hunk 4]
```diff
@@ -45,6 +54,22 @@ def __init__(self, config=None, backend=None) -> None:
             api_key = config.get("bugzilla").get("api_key")
             self.backend = Bugzilla(url, api_key=api_key)
 
+    def add_comment(self, issue, comment):
```
Comment (jguiditta) [inline src/bugjira/broker.py]: Does this need the type hint, or do we skip because it is declared in the parent?

**Review by jguiditta — COMMENTED** [review]
Minor question, otherwise looks good to me!

File: src/bugjira/broker.py (hunk 4 above)
Comment (eggmaster) [reply src/bugjira/broker.py]: good catch, I'll add for consistency's sake

Comment (jguiditta) [reply src/bugjira/broker.py]: ACK, looks good!

**Review by jguiditta — APPROVED** [review]

**Review by jpichon — APPROVED** [review]

**Pull request merged**